crates/store/src/db/mod.rs

```diff
@@ -548,36 +492,28 @@ impl Db {
         block: ProvenBlock,
         notes: Vec<(NoteRecord, Option<Nullifier>)>,
     ) -> Result<()> {
-        self.pool
-            .get()
-            .await?
-            .interact(move |conn| -> Result<()> {
-                // TODO: This span is logged in a root span, we should connect it to the parent one.
-                let _span = info_span!(target: COMPONENT, "write_block_to_db").entered();
-
-                let transaction = conn.transaction()?;
-                sql::apply_block(
-                    &transaction,
-                    block.header(),
-                    &notes,
-                    block.created_nullifiers(),
-                    block.updated_accounts(),
-                    block.transactions(),
-                )?;
-
-                let _ = allow_acquire.send(());
-                acquire_done.blocking_recv()?;
-
-                transaction.commit()?;
-
-                Ok(())
-            })
-            .await
-            .map_err(|err| {
-                DatabaseError::InteractError(format!("Apply block task failed: {err}"))
-            })??;
-
-        Ok(())
+        self.transact("apply block", move |conn| -> Result<()> {
+            // TODO: This span is logged in a root span, we should connect it to the parent one.
+            let _span = info_span!(target: COMPONENT, "write_block_to_db").entered();
```

Comment:
I think we would just have the transact fns span handle this instead of manually doing a span and event here. So would suggest removing this line and add an `#[instrument` line to 
```
    pub(crate) async fn transact<R, E, Q, M>(&self, msg: M, query: Q) -> std::result::Result<R, E>
```

This would be a trace or debug level span probably.

Reply:
I think we can probably just delete this one. It probably came from a time before we instrumented at the top of this function.

I'm somewhat in favor of instrumenting `transact` but at the same time its almost always just a single call nested inside another instrumented function so its probably unnecessary?